frame/composable-traits/src/lending/mod.rs

```diff
@@ -97,17 +114,112 @@ impl<LiquidationStrategyId, AssetId: Copy> CreateInput<LiquidationStrategyId, As
 	}
 }
 
-#[derive(Encode, Decode, Default, TypeInfo)]
+#[derive(Encode, Decode, Default, TypeInfo, Debug)]
 pub struct MarketConfig<VaultId, AssetId, AccountId, LiquidationStrategyId> {
+	/// The owner of this market.
 	pub manager: AccountId,
-	pub borrow: VaultId,
-	pub collateral: AssetId,
+	/// The vault containing the borrow asset.
+	pub borrow_asset_vault: VaultId,
+	/// The asset being used as collateral.
+	pub collateral_asset: AssetId,
 	pub collateral_factor: MoreThanOneFixedU128,
 	pub interest_rate_model: InterestRateModel,
-	pub under_collaterized_warn_percent: Percent,
+	pub under_collateralized_warn_percent: Percent,
 	pub liquidators: Vec<LiquidationStrategyId>,
 }
 
+/// Different ways that a market can be repaid.
+// REVIEW: Perhaps add an "interest only" strategy?
+// InterestOnly
+#[derive(Encode, Decode, TypeInfo, Debug, Clone, PartialEq)]
+pub enum RepayStrategy<T> {
```

Comment:
I think that manager of pool may set one of 2 strategies. Either force to repay interest asap, and only if something remains, repay principal. Or repay part of interest and part of principal. Examples. Your principal is 100 and accrued interest is 10. You pay 11. Either you stay with 99 principla after repay. Or with 90 principal and 10 interest.
Repaing only interest weird, you can have huge debts, repay principal and has free loan (because we do not accrue on interest). 
I do not think that concept of repaying full or by parts is meaningull. User just pays, and gets what market owner decided and amount he put into.